mlserver/grpc/interceptors.py

```diff
@@ -0,0 +1,20 @@
+from typing import Callable, Awaitable
+
+from grpc.aio import ServerInterceptor
+from grpc import HandlerCallDetails, RpcMethodHandler
+
+from .logging import logger
+
+
+class LoggingInterceptor(ServerInterceptor):
+    def _get_log_message(self, handler_call_details: HandlerCallDetails) -> str:
+        return handler_call_details.method
+
+    async def intercept_service(
+        self,
+        continuation: Callable[[HandlerCallDetails], Awaitable[RpcMethodHandler]],
+        handler_call_details: HandlerCallDetails,
+    ) -> RpcMethodHandler:
+        logger.info(self._get_log_message(handler_call_details))
```

Comment:
just for my reference, what is an example of `handler_call_details.method` ?

also do we have any potential issues with writing async to the same log file? especially with `ProcessThreadExecutor` ?

Reply:
I realise that `ProcessThreadExecutor` is lower in the call stack so probably not having an effect here in the grpc server.

Reply:
That's a good point, not sure what would happen with the logger on the inference workers. What would `spawn` do in this case? Would it just replicate the standard output handler?

On the former point, `handler_call_details.method` would be a string like `inference.GRPCInferenceService/ModelInfer` (i.e. `<proto package>.<service name>/<service method>`). Or at least that's what I found out while experimenting!